src/main/main-directive.js

```diff
@@ -2,12 +2,21 @@
     'use strict';
 
     angular.module('stockflux.main')
-        .directive('main', [() => {
+        .directive('main', ['$timeout', ($timeout) => {
             return {
                 restrict: 'E',
                 templateUrl: 'main/main.html',
                 controller: 'MainCtrl',
-                controllerAs: 'mainCtrl'
+                controllerAs: 'mainCtrl',
+                link: (scope, element) => {
+                    scope.$on('compactChanging', () => {
+                        element.addClass('disable-transitions');
+
+                        $timeout(() => {
+                            element.removeClass('disable-transitions');
+                        }, 100);
```

Comment:
I wonder if there should be two events: `before-compact-change` and `after-compact-change`?

That way you might not need the `$timeout`, and you can broadcast on either side of the compact change in the toolbar-controller. Or do async things mess that up?